presto-main/src/main/java/io/prestosql/operator/scalar/JoniRegexpFunctions.java

```diff
@@ -306,4 +306,74 @@ private static void validateGroup(long group, Region region)
             throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Pattern has %d groups. Cannot access group %d", region.numRegs - 1, group));
         }
     }
+
+    @ScalarFunction
+    @Description("returns the beginning position of the matched substring")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source, @SqlType(JoniRegexpType.NAME) Regex pattern)
+    {
+        return regexpPosition(source, pattern, 0);
+    }
+
+    @ScalarFunction
+    @Description("returns the beginning position of the matched substring from the specified starting position")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source,
+                                      @SqlType(JoniRegexpType.NAME) Regex pattern,
+                                      @SqlType(StandardTypes.INTEGER) long start)
+    {
+        Matcher matcher = pattern.matcher(source.getBytes());
+        return matcher.search((int) start, source.length(), Option.DEFAULT);
+    }
+
+    @ScalarFunction
+    @Description("returns the beginning position of the nth occurrence substring from the specified starting position")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source,
+                                      @SqlType(JoniRegexpType.NAME) Regex pattern,
+                                      @SqlType(StandardTypes.INTEGER) long start,
+                                      @SqlType(StandardTypes.INTEGER) long occurrence)
+    {
+        Matcher matcher = pattern.matcher(source.getBytes());
+
+        long count = 0;
+        int iStart = (int) start;
+        while (true) {
+            int offset = matcher.search(iStart, source.length(), Option.DEFAULT);
+            if (offset == -1 || (iStart == matcher.getBegin())) {
```

Comment:
What `(iStart == matcher.getBegin()` is used for? what if you want second occurence of `abc` in `abcabc`?

Reply:
In case pattern string is empty